### PR TITLE
Checkout attributes

### LINKED ISSUE #92: Add note-attribute to BUYCheckout
Is it possible?

### PR DESCRIPTION
**Pull request by runmad**
This adds checkout attributes and fixes #92.

Includes serialization conversion and unit tests.

@davidmuzi as discussed, please have a look.

@bgulanowski this affects some of your work.

### CODE REVIEW

**Comment by bgulanowski** [comment]
:+1:

File: Mobile Buy SDK/Mobile Buy SDK/Models/BUYCheckoutAttribute.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
+//
+//  BUYCheckoutAttribute.h
+//  Mobile Buy SDK
+//
+//  Created by Shopify.
+//  Copyright (c) 2015 Shopify Inc. All rights reserved.
```
Comment (davidmuzi) [inline Mobile Buy SDK/Mobile Buy SDK/Models/BUYCheckoutAttribute.h]: 2016

**Comment by bgulanowski** [comment]
I just looked at this again, and now that I'm paying attention, I'm wishing `CheckoutAttribute` did not exist as a new entity. We can easily create a dictionary property on the Checkout itself.

**Comment by runmad** [comment]
I think that's fair to change in terms of how it should be consumed. Feel free to change this.

**Pull request merged**